Lib/test/test_symtable.py

```diff
@@ -87,13 +88,14 @@ def test_type(self):
         self.assertEqual(self.internal.get_type(), "function")
         self.assertEqual(self.foo.get_type(), "function")
         self.assertEqual(self.Alias.get_type(), "type alias")
-        self.assertEqual(self.GenericAlias.get_type(), "type parameter")
+        self.assertEqual(self.GenericAlias.get_type(), "type parameters")
```

Comment:
Should I do it for the backports only? or should I also mention it in 3.13 and 3.14?

Reply:
Here is what I suggest:

- 3.12: use enumeration members with *old* values (to avoid breakage) but still use enumeration values. use `.. versionchanged:: 3.12` for this specific version.

- 3.13, 3.14: use enumeration members with *new* values (both versions are not stable so we should be able to use whatever values we want). Use `.. versionchanged:: 3.13` instead of `.. versionchanged:: 3.13` for both verions.

More precisely, here is what the RST would be:

**For 3.12**

```diff
+   .. attribute:: TYPE_ALIAS
+      :value: "type alias"
(...)
+   .. attribute:: TYPE_PARAMETERS
+      :value: "type parameter"
(...)
+
+   .. attribute:: TYPE_VARIABLE
+      :value: "TypeVar bound"
(...)
-      Return the type of the symbol table.  Possible values are ``'class'``,
-      ``'module'``, ``'function'``, ``'annotation'``, ``'TypeVar bound'``,
-      ``'type alias'``, and ``'type parameter'``. The latter four refer to
-      different flavors of :ref:`annotation scopes <annotation-scopes>`.
+      Return the type of the symbol table.  Possible values are members
+      of the :class:`SymbolTableType` enumeration.

       .. versionchanged:: 3.12
          Added ``'annotation'``,  ``'TypeVar bound'``, ``'type alias'``,
          and ``'type parameter'`` as possible return values.

+      .. versionchanged:: 3.12
+         Return values are members of the :class:`SymbolTableType` enumeration.
+
+         The exact values of the returned string may change in the future,
+         and thus, it is recommended to use :class:`SymbolTableType` members
+         instead of hard-coded strings.
```

**For 3.13 and 3.14**

```diff
+   .. attribute:: TYPE_ALIAS
+      :value: "type alias"
(...)
+   .. attribute:: TYPE_PARAMETERS
+      :value: "type parameter"
(...)
+
+   .. attribute:: TYPE_VARIABLE
+      :value: "type variable"
(...)
-      Return the type of the symbol table.  Possible values are ``'class'``,
-      ``'module'``, ``'function'``, ``'annotation'``, ``'TypeVar bound'``,
-      ``'type alias'``, and ``'type parameter'``. The latter four refer to
-      different flavors of :ref:`annotation scopes <annotation-scopes>`.
+      Return the type of the symbol table.  Possible values are members
+      of the :class:`SymbolTableType` enumeration.

       .. versionchanged:: 3.12
          Added ``'annotation'``,  ``'TypeVar bound'``, ``'type alias'``,
          and ``'type parameter'`` as possible return values.

+      .. versionchanged:: 3.13
+         Return values are members of the :class:`SymbolTableType` enumeration.
+
+         The exact values of the returned string may change in the future,
+         and thus, it is recommended to use :class:`SymbolTableType` members
+         instead of hard-coded strings.
```

And in the tests, we would check that separately depending on the branch we are working on.

Reply:
I'd prefer to leave 3.12 unchanged; adding an enum seems too much in a bugfix release, and the bug this issue fixes isn't as bad there.